### PR TITLE
Update JSON-LD for new authentication means

### PR DESCRIPTION
**Pull request by woutslakhorst**
added assuranceLevel to AuthCred, added NutsEmplCred and SelfSignedPresentation

part of #2047 

- assuranceLevel is modelled after https://openid.net/specs/openid-connect-4-identity-assurance-1_0.html
- Example on how to use on [playground](https://json-ld.org/playground/#startTab=tab-nquads&json-ld=%7B%22%40context%22%3A%5B%22https%3A%2F%2Fwww.w3.org%2F2018%2Fcredentials%2Fv1%22%2C%22https%3A%2F%2Fw3id.org%2Fsecurity%2Fsuites%2Fjws-2020%2Fv1%22%2C%7B%22%40version%22%3A1.1%2C%22%40protected%22%3Atrue%2C%22id%22%3A%22%40id%22%2C%22type%22%3A%22%40type%22%2C%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22nuts%22%3A%22https%3A%2F%2Fnuts.nl%2Fcredentials%2Fv1%23%22%2C%22NutsSelfSignedPresentation%22%3A%22nuts%3ANutsSelfSignedPresentation%22%2C%22NutsEmployeeCredential%22%3A%7B%22%40id%22%3A%22nuts%3ANutsEmployeeCredential%22%2C%22%40context%22%3A%7B%22%40version%22%3A1.1%2C%22%40propagate%22%3Atrue%2C%22%40protected%22%3Atrue%2C%22id%22%3A%22%40id%22%2C%22type%22%3A%22%40type%22%2C%22identifier%22%3A%22schema%3Aidentifier%22%2C%22roleName%22%3A%22schema%3AroleName%22%2C%22member%22%3A%7B%22%40id%22%3A%22schema%3Amember%22%2C%22%40context%22%3A%7B%22initials%22%3A%22schema%3Ainitials%22%2C%22familyName%22%3A%22schema%3AfamilyName%22%7D%7D%7D%7D%7D%5D%2C%22type%22%3A%5B%22VerifiablePresentation%22%2C%22NutsSelfSignedPresentation%22%5D%2C%22verifiableCredential%22%3A%5B%7B%22type%22%3A%5B%22VerifiableCredential%22%2C%22NutsEmployeeCredential%22%5D%2C%22issuer%22%3A%22did%3Anuts%3A123456789%22%2C%22expirationDate%22%3A%222023-04-03T20%3A34%3A17.687862%2B01%3A00%22%2C%22credentialSubject%22%3A%7B%22type%22%3A%22schema%3AOrganization%22%2C%22id%22%3A%22did%3Anuts%3Aa%22%2C%22identifier%22%3A%22481%22%2C%22roleName%22%3A%22Verpleegkundige%20niveau%202%22%2C%22member%22%3A%7B%22type%22%3A%22schema%3APerson%22%2C%22initials%22%3A%22J%22%2C%22familyName%22%3A%22van%20Dijk%22%7D%7D%7D%5D%2C%22proof%22%3A%7B%22challenge%22%3A%22LOGIN%20CONTRACT%22%2C%22type%22%3A%22JsonWebSignature2020%22%2C%22verificationMethod%22%3A%22did%3Anuts%3A123456789%23key-1%22%2C%22created%22%3A%222023-04-03T16%3A34%3A17.687862%2B01%3A00%22%2C%22jws%22%3A%22eyJhbGciOiJFUzI1NiIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..hKcboC8m6YnZPi6ReJAYs0J0Ztn5nxcx2EavoXdtrkWxmE1JZmImW89_8IIgjvfI8XtGeDlEnGywAuY2u7y9Bw%22%7D%7D&frame=%7B%7D&context=%7B%7D)

### CODE REVIEW

**Comment by reinkrul** [comment]
Sloppy PR title

**Review by reinkrul — CHANGES_REQUESTED** [review]
Looking at the playground and the context, `identifier` is now a property of the `organization`, I don't think that's correct? (and now it has both `id` and `identifier`). Shouldn't it be a property of `member`? Same goes for `roleName` i.m.o.

**Comment by reinkrul** [comment]
What is the rationale behind the `SelfSignedPresentation` (concept-wise)?

**Comment by stevenvegt** [comment]
The `credentialSubject` is the same as the `issuer` and of type `schema:Organization` which is correct, otherwise it would not be selfSigned and the presentation signature would not prove the possession.

However, by putting properties like `identifier` and `roleName` on this level, this credential claims that the care Org has a `roleName` and `identifier` which is false. You need to keep the structure as defined by schema.org about organizations: https://schema.org/Organization and use the `member` property to include the role, see this example: https://schema.org/Organization#eg-0206.

The credential would than become:
```json
{
  "issuer": "did:nuts:careOrg",
  "type": ["VerifiableCredential", "NutsEmployeeCredential"],
  "expirationDate": "2023-04-03T20:34:17.687862+01:00",
  "credentialSubject": {
    "type":"Organization",
    "id":"did:nuts:careOrg",
    "member":{
      "type":"EmployeeRole",
      "identifier": "481",
      "roleName": "Verpleegkundige niveau 2",
      "member": {
        "type": "Person",
        "initials": "J",
        "familyName": "van Dijk",
        "email": "j.vandijk@example.com"
      }
    }
  }
}
```

Screenshot 2023-04-13 at 10 31 07

**Comment by reinkrul** [comment]
> However, by putting properties like `identifier` and `roleName` on this level, this credential claims that the care Org has a `roleName` and `identifier` which is false. You need to keep the structure as defined by schema.org about organizations: https://schema.org/Organization and use the `member` property to include the role, see this example: https://schema.org/Organization#eg-0206.

Should we use `id` for member instead of `identifier`? Or `id` with prefixing the organization ID?, e.g. `did:nuts:orgA/member/461`?

**Comment by stevenvegt** [comment]
It also seems that the types `EmployeeRole`, `Person` and `Organization` must be defined in the schema, or must be prefixed with `schema` in order to work.

**Comment by stevenvegt** [comment]
Since the credential only uses fields from schema.org we can make the context a lot simpler by just setting the `@vocab` to 'schema` and every field automatically gets the correct prefix:
```
     "NutsEmployeeCredential": {
        "@id": "nuts:NutsEmployeeCredential",

        "@context": {
          "@version": 1.1,
          "@propagate": true,
          "@protected": true,
          "@vocab":"schema"
        }
      }
```

See [this example on the playground](https://json-ld.org/playground/#startTab=tab-table&json-ld=%7B%22%40context%22%3A%5B%22https%3A%2F%2Fwww.w3.org%2F2018%2Fcredentials%2Fv1%22%2C%22https%3A%2F%2Fw3id.org%2Fsecurity%2Fsuites%2Fjws-2020%2Fv1%22%2C%7B%22%40version%22%3A1.1%2C%22%40protected%22%3Atrue%2C%22id%22%3A%22%40id%22%2C%22type%22%3A%22%40type%22%2C%22schema%22%3A%22http%3A%2F%2Fschema.org%2F%22%2C%22nuts%22%3A%22https%3A%2F%2Fnuts.nl%2Fcredentials%2Fv1%23%22%2C%22NutsSelfSignedPresentation%22%3A%22nuts%3ANutsSelfSignedPresentation%22%2C%22NutsEmployeeCredential%22%3A%7B%22%40id%22%3A%22nuts%3ANutsEmployeeCredential%22%2C%22%40context%22%3A%7B%22%40version%22%3A1.1%2C%22%40propagate%22%3Atrue%2C%22%40protected%22%3Atrue%2C%22%40vocab%22%3A%22schema%22%7D%7D%7D%5D%2C%22type%22%3A%5B%22VerifiablePresentation%22%2C%22NutsSelfSignedPresentation%22%5D%2C%22verifiableCredential%22%3A%5B%7B%22type%22%3A%5B%22VerifiableCredential%22%2C%22NutsEmployeeCredential%22%5D%2C%22issuer%22%3A%22did%3Anuts%3A123456789%22%2C%22expirationDate%22%3A%222023-04-03T20%3A34%3A17.687862%2B01%3A00%22%2C%22credentialSubject%22%3A%7B%22%40type%22%3A%22Organization%22%2C%22id%22%3A%22did%3Anuts%3A123456789%22%2C%22member%22%3A%7B%22identifier%22%3A%22481%22%2C%22type%22%3A%22EmployeeRole%22%2C%22roleName%22%3A%22Verpleegkundige%20niveau%202%22%2C%22member%22%3A%7B%22type%22%3A%22Person%22%2C%22initials%22%3A%22J%22%2C%22familyName%22%3A%22van%20Dijk%22%2C%22foo%22%3A%22bar%22%7D%7D%7D%7D%5D%2C%22proof%22%3A%7B%22challenge%22%3A%22LOGIN%20CONTRACT%22%2C%22type%22%3A%22JsonWebSignature2020%22%2C%22verificationMethod%22%3A%22did%3Anuts%3A123456789%23key-1%22%2C%22created%22%3A%222023-04-03T16%3A34%3A17.687862%2B01%3A00%22%2C%22jws%22%3A%22eyJhbGciOiJFUzI1NiIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..hKcboC8m6YnZPi6ReJAYs0J0Ztn5nxcx2EavoXdtrkWxmE1JZmImW89_8IIgjvfI8XtGeDlEnGywAuY2u7y9Bw%22%7D%7D).

**Comment by stevenvegt** [comment]
> > However, by putting properties like `identifier` and `roleName` on this level, this credential claims that the care Org has a `roleName` and `identifier` which is false. You need to keep the structure as defined by schema.org about organizations: https://schema.org/Organization and use the `member` property to include the role, see this example: https://schema.org/Organization#eg-0206.
> 
> Should we use `id` for member instead of `identifier`? Or `id` with prefixing the organization ID?, e.g. `did:nuts:orgA/member/461`?

An `id` is defined as an `@id` and should only contain an IRI. So, a simple string or number is not sufficient. Your suggestion to add a prefix makes sense, but could also be a case of YAGNI. If you want to search for this employeeCredential you can search for _orgs with DID `did:nuts:123` which have a member of type `EmployeeRole` with identity `123`_

**Review by stevenvegt — APPROVED** [review]
Not sure why you removed the line 37-38 and 61, but the additions look good to me.

File: vcr/assets/assets/contexts/nuts.ldjson [hunk 1]
```diff
@@ -34,8 +34,6 @@
 
         "id": "@id",
         "type": "@type",
-        "cred": "https://www.w3.org/2018/credentials#",
-        "nuts": "https://nuts.nl/credentials/v1#",
```
Comment (gerardsn) [inline vcr/assets/assets/contexts/nuts.ldjson:38]: does line 29 use this?

**Review by reinkrul — APPROVED** [review]
I think it looks okay. We have some JSON-LD operation tests to test our contexts in `jsonld/testsuite/compat_test.go`, maybe worth adding a test case

File: vcr/assets/assets/contexts/nuts.ldjson (hunk 1 above)
Comment (woutslakhorst) [reply vcr/assets/assets/contexts/nuts.ldjson:38]: defined at highest level. Seems to propagate.

**Pull request merged**